beetsplug/parentwork.py

```diff
@@ -0,0 +1,225 @@
+            if artist['type'] == 'composer':
+                if 'end' in artist.keys():
+                    work_date = artist['end']
+
+    if 'work-relation-list' in work_info['work']:
+        for work_father in work_info['work']['work-relation-list']:
+            if work_father['type'] == 'parts' \
+                    and work_father.get('direction') == 'backward':
+                father_id = work_father['work']['id']
+                return father_id, work_date
+    return None, work_date
+
+
+def work_parent(work_id):
+    """This function finds the parentwork id of a work given its id. """
+    work_date = None
+    while True:
+        (new_work_id, work_date) = work_father(work_id, work_date)
+        if not new_work_id:
+            return work_id, work_date
+            break
+        work_id = new_work_id
+    return work_id, work_date
+
+
+def find_parentwork(work_id):
+    """This function gives the work relationships (dict) of a parent_work
+    given the id of the work"""
+    parent_id, work_date = work_parent(work_id)
+    work_info = musicbrainzngs.get_work_by_id(parent_id,
+                                              includes=["artist-rels"])
+    return work_info, work_date
+
+
+class ParentWorkPlugin(BeetsPlugin):
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+
+        self.config.add({
+            'auto': False,
+            'force': False,
+        })
+
+        self._command = ui.Subcommand(
+            'parentwork',
+            help=u'Fetches parent works, composers and dates')
+
+        self._command.parser.add_option(
+            u'-f', u'--force', dest='force',
+            action='store_true', default=None,
+            help=u'Re-fetches all parent works')
+
+        if self.config['auto']:
+            self.import_stages = [self.imported]
+
+    def commands(self):
+
+        def func(lib, opts, args):
+            self.config.set_args(opts)
+            force_parent = self.config['force'].get(bool)
+            write = ui.should_write()
+
+            for item in lib.items(ui.decargs(args)):
+                self.find_work(item, force_parent)
+                item.store()
+                if write:
+                    item.try_write()
+
+        self._command.func = func
+        return [self._command]
+
+    def imported(self, session, task):
+        """Import hook for fetching parent works automatically.
+        """
+        force_parent = self.config['force'].get(bool)
+
+        for item in task.imported_items():
+            self.find_work(item, force_parent)
+            item.store()
+
+    def get_info(self, item, work_info, parent_composer, parent_composer_sort,
+                 parent_work, parent_work_disambig, parent_work_id,
+                 composer_ids):
+        """Given the parentwork info dict, this function updates
+        parent_composer, parent_composer_sort, parent_work,
+        parent_work_disambig, work_ids and composer_ids"""
```

Comment:
I'd prefer for this function to *return* information rather than *modifying* its arguments. The caller can then update all the arrays to collect the data across multiple works.